bittensor/_neuron/text/template_server/run.py

```diff
@@ -101,15 +104,61 @@ def blacklist(pubkey:str, request_type:bittensor.proto.RequestType) -> bool:
         """
         # Check for registrations
 
-        is_registered = pubkey in metagraph.hotkeys
-        if not is_registered:
-            if config.neuron.blacklist_allow_non_registered:
+        def registration_check():
+            # If we allow non-registered requests return False = not blacklisted.
+            is_registered = pubkey in metagraph.hotkeys
+            if not is_registered:
+                if config.neuron.blacklist_allow_non_registered:
+                    
+                    return False
+                raise Exception('blacklist')
+
+        # Check for stake
+        def stake_check() -> bool:
+                
+            # Check stake.
+            uid = metagraph.hotkeys.index(pubkey)
+            if metagraph.S[uid].item() < config.neuron.blacklist.stake:
+                raise Exception('Stake blacklist')
+            return False
+
+        def validator_check():
+
+            uid = metagraph.hotkeys.index(pubkey)
+            if (metagraph.W[uid] >0).sum() ==n_topk_peer_weights:
+                return False
+
+            raise Exception('Validator blacklist')
+
+
+        # Check for time
+        def time_check():
+            current_time = datetime.now()
+            if pubkey in timecheck.keys():
+                prev_time = timecheck[pubkey]
+                if current_time - prev_time >= timedelta(seconds=config.neuron.blacklist.time):
+                    timecheck[pubkey] = current_time
+                    return False
+                else:
+                    timecheck[pubkey] = current_time
+                    raise Exception('blacklist')
+            else:
+                timecheck[pubkey] = current_time
                 return False
+
+        # Black list or not
+        try:
+            registration_check()
+
+            stake_check()
+
+            validator_check()
             
+            return False
+
+        except Exception as e:
```

Comment:
the test would not bug you but ala might hahaha

Reply:
haha this is to catch exceptions from the tests, so tests can run in sequential. I think it should be fine?

Reply:
ugh. will allow.